bundles/cif-virtual-catalog/src/main/java/com/adobe/cq/commerce/virtual/catalog/data/impl/ProductBindingCreator.java

```diff
@@ -0,0 +1,129 @@
+
+    private ResourceResolver resolver;
+
+    @Reference
+    private ResourceResolverFactory resolverFactory = null;
+
+    protected void activate(final ComponentContext context) throws LoginException {
+        LOG.debug("Activating the component");
+        final Map<String, Object> map = new HashMap<>();
+        map.put(ResourceResolverFactory.SUBSERVICE, PRODUCT_BINDING_SERVICE);
+        resolver = resolverFactory.getServiceResourceResolver(map);
+        LOG.debug("Do we have a resolver? {}", resolver != null);
+    }
+
+    protected void deactivate() {
+        resolver.close();
+    }
+
+    @Override
+    public void onChange(List<ResourceChange> list) {
+        LOG.debug("Change detected somewhere...");
+        list.stream().filter(change -> {
+            String path = change.getPath();
+            LOG.debug("Processing path {}", path);
+            return !path.endsWith(JcrConstants.JCR_CONTENT) && path.contains(Constants.COMMERCE_BUCKET_PATH);
+        }).forEach(change -> {
+            if (change.getType() == ResourceChange.ChangeType.ADDED) {
+                processAddition(change);
+            }
+        });
+    }
+
+    private void processAddition(ResourceChange change) {
+        String path = change.getPath();
+        LOG.debug("Process resource addition at path {}", path);
+
+        Resource changedResource = resolver.getResource(path);
+        Resource contentResource = changedResource.getChild(JcrConstants.JCR_CONTENT);
+
+        ValueMap properties;
+        if (contentResource != null) {
+            properties = contentResource.getValueMap();
+        } else {
+            properties = changedResource.getValueMap();
+        }
+
+        String storeView = properties.get(Constants.PN_MAGENTO_STORE, "");
+        if (StringUtils.isEmpty(storeView)) {
+            LOG.warn("The configuration at path {} doesn't have a '{}' property", path, Constants.PN_MAGENTO_STORE);
+            return;
+        }
+
+        String configRoot = path.substring(0, path.indexOf(Constants.COMMERCE_BUCKET_PATH) - 1);
+        String configName = configRoot.substring(configRoot.lastIndexOf("/") + 1);
+
+        String bindingName = configName + "-" + storeView;
+        LOG.debug("New binding name: {}", bindingName);
+
+        Map<String, Object> mappingProperties = new HashMap<>();
+        mappingProperties.put(JcrConstants.JCR_PRIMARYTYPE, "sling:Folder");
+        mappingProperties.put(JcrConstants.JCR_TITLE, bindingName);
+        mappingProperties.put(Constants.PN_CONF, configRoot);
+
+        Resource parent = resolver.getResource(BINDINGS_PARENT_PATH);
+        if (parent == null) {
+            LOG.warn("Binding parent path not found at {}. Nothing to do here...", BINDINGS_PARENT_PATH);
+            return;
+        }
+
+        try {
+            LOG.debug("Creating a new resource at {}", parent.getPath() + "/" + bindingName);
+            resolver.create(parent, bindingName, mappingProperties);
```

Comment:
Shouldn't you check if the binding already exists? It can be that existing projects already added the binding nodes to their project content package to automatically create the bindings, I know for sure that some projects did that because they reported some issues about that.